core/query/src/main/java/datawave/core/query/configuration/GenericQueryConfiguration.java

```diff
@@ -219,10 +291,32 @@ public boolean canRunQuery() {
         }
 
         // At least one QueryData was provided
-        if (null == this.queries) {
+        if (null == this.getQueriesIter()) {
             return false;
         }
 
         return true;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        GenericQueryConfiguration that = (GenericQueryConfiguration) o;
+        return isCheckpointable() == that.isCheckpointable() && getBaseIteratorPriority() == that.getBaseIteratorPriority()
```

Comment:
Probably need to make sure all of the class members are accounted for in equals and hashcode since this branch has been around as long as it has.

Reply:
Another alternative is Apache's EqualsBuilder and HashCodeBuilder, we've used those elsewhere in the project.

Reply:
@alerman unfortunately, for some of these classes i think there are class members which we *don't* want to consider when comparing for equality.